uwcsc/codeybot
# Convert resume pdfs to images

# Summary of Changes
PDFs sent to the #resumes channel will be converted into images automatically by Codey.

## TODOs Before Merging
- We need to update the staging and production dockerfiles to include these lines that download deps necessary for the PDF-\>image converter:
```
RUN apk add --update ghostscript
RUN apk add --update graphicsmagick
```
- We need to update the staging and production configs to account for the new config variable, `RESUME_CHANNEL_ID` (see below)

# Testing Notes
1. Add the `RESUME_CHANNEL_ID` variable to your `config.json`; this is the channel where the resumes are sent (in this case it will be #resumes)
2. Send a message without any attachment: nothing should happen (this is similar to when people post google drive links to their resumes)
3. Send a message with a pdf of a resume: Codey should automatically send back an image corresponding to the resume.

## Screenshots

Nothing happens if no attachment is sent, so the screenshot for the case where a resume pdf is attached is shown below:

![image](https://user-images.githubusercontent.com/76544623/183744231-76cba0c9-3ddf-49d6-a28c-0e6b58b5c2f6.png)
